### ISSUE TITLE
Sorting out outputted paths

### ISSUE THREAD

**Issue by palfrey** [swagger-autogen/swagger-autogen#238]
We recently had an issue where one of our devs was getting different outputs from swagger-autogen on their machine from others. I think that the problem is the outputted `paths` aren't explicitly sorted (AFAIK) and so depend on the "natural" sort order being stable for repeatable output. Could we instead sort the output paths so we know they're always in the same order?

Happy to provide a PR to fix this if wanted, but wanted to get feedback beforehand to check this didn't cause other issues.

**Comment by davibaltar** [comment]
Hi @palfrey. Thank you for opening the issue. 

I've been thinking about it. I've started to implement an option to sort parameters (but it's still in test, [see here](https://github.com/swagger-autogen/swagger-autogen/blob/master/swagger-autogen.js#L27)). However, sort paths will be one of my next steps. 

In the version 3.x it would be possible to change the default order, but in the version 2.x it's not possible to change the natural order as a default, because we need to ensure the same pattern/compatibility for all 2.x versions. 

About options to sort the paths, I've thought two ways:

1. An option to sort all the paths, for example: 
```js
const swaggerAutogen = require('swagger-autogen')({ openapi: '3.0.0', sortPaths: 'alphabetically' });
```

2. A new swagger tag to ensure the position of the path, such as `#swagger.order = 1` or `#swagger.pathOrder = 1` where `1` is the position. I'm not sure about the tag name yet. From what I understand, this is your need.

Did you think anything different from that?

**Comment by palfrey** [comment]
> I've been thinking about it. I've started to implement an option to sort parameters (but it's still in test, [see here](https://github.com/swagger-autogen/swagger-autogen/blob/master/swagger-autogen.js#L27)). However, sort paths will be one of my next steps.

👍🏻 

> In the version 3.x it would be possible to change the default order, but in the version 2.x it's not possible to change the natural order as a default, because we need to ensure the same pattern/compatibility for all 2.x versions.

It should however be possible to add a sort order option to 2.x right? Default to the natural order, but optionally set an order? That would at least allow me to fix issues with my codebase :)

>     1. An option to sort all the paths, for example:
> 
> ```js
> const swaggerAutogen = require('swagger-autogen')({ openapi: '3.0.0', sortPaths: 'alphabetically' });
> ```

I was thinking of something like this.

**Comment by davibaltar** [comment]
> It should however be possible to add a sort order option to 2.x right? Default to the natural order, but optionally set an order? That would at least allow me to fix issues with my codebase :)

yes @palfrey! I'm going to mark it as a new feature and news about it, I'll post it here.
